scipy/interpolate/rbf.py

```diff
@@ -59,6 +59,8 @@ class Rbf:
     A class for radial basis function interpolation of functions from
     N-D scattered data to an M-D domain.
 
+    `Rbf` is legacy code, for new usage please use `RBFInterpolator` instead.
+
```

Comment:
I would either use a note or a warning here (`.. note::`). We could also add it to the list of deprecation. What do you think @rgommers? I see we have a `_deprecated`.

Reply:
@rgommers commented on this on the mailing list (https://mail.python.org/pipermail/scipy-dev/2021-February/024586.html). I will change this to an RST note for now.

Reply:
Oh indeed I missed this email. It's ok then 👍